pkg/transforms/mqtt_test.go

```diff
@@ -0,0 +1,114 @@
+package transforms
+
+import (
+	"strconv"
+	"strings"
+	"testing"
+
+	MQTT "github.com/eclipse/paho.mqtt.golang"
+	"github.com/edgexfoundry/app-functions-sdk-go/pkg/excontext"
+	logger "github.com/edgexfoundry/go-mod-core-contracts/clients/logging"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/stretchr/testify/assert"
+)
+
+var addr models.Addressable
+
+func init() {
+	lc = logger.NewClient("app_functions_sdk_go", false, "./test.log", "DEBUG")
+	addr = models.Addressable{
+		Address:   "localhost",
+		Port:      1883,
+		Protocol:  "tcp",
+		Publisher: "",
+		Password:  "",
+		Topic:     "testMQTTTopic",
+	}
+}
+
+func TestMQTTSend(t *testing.T) {
+	t.SkipNow()
+	protocol := strings.ToLower(addr.Protocol)
+	opts := MQTT.NewClientOptions()
+	broker := protocol + "://" + addr.Address + ":" + strconv.Itoa(addr.Port) + addr.Path
+	opts.AddBroker(broker)
+	opts.SetClientID(addr.Publisher)
+	opts.SetUsername(addr.User)
+	opts.SetPassword(addr.Password)
+	opts.SetAutoReconnect(false)
+
+	sender := MQTTSender{
+		client: MQTT.NewClient(opts),
+		topic:  "",
+	}
+	ctx := excontext.Context{
+		LoggingClient: lc,
+	}
+	dataToSend := "SOME DATA TO SEND"
+	continuePipeline, result := sender.MQTTSend(ctx, dataToSend)
```

Comment:
Does this unit test require MQTT service to be running?

Reply:
For all of successful test cases...yes. But i do call t.SkipNow() in the ones that require it. So the tests are written, but currently wont run either unless you comment out the skipnow(). They currently dont have mqtt_test.go in the export distro, so I'm guessing a mock needs to happen.